### PR TITLE
add an error when buildOutput fails

### LINKED ISSUE #156: [🐛 Bug]: buildSuccess check crushes with bad error message
### Cli version

0.7.0

### Description

In the following check:
https://github.com/cloudflare/next-on-pages/blob/9834066c889453bd1d3feb620892209c45a3482a/src/buildApplication/buildApplication.ts#L28-L38

added in PR #145 

`err` can be `undefined` causing the following error:
Screenshot 2023-04-06 at 23 14 55

we should use optional chaining and provide a nice default error message instead of crushing

### PR DESCRIPTION
**Pull request by abhishek276533**
Fix #156

### CODE REVIEW

**Comment by abhishek276533** [comment]
Not able to add Reviewers @dario-piotrowicz

**Comment by dario-piotrowicz** [comment]
> Not able to add Reviewers @dario-piotrowicz

I've added myself 🙂👍 

(but the PR is still in draft so it's not ready to be reviewed right?)

File: .changeset/modern-keys-mix.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@cloudflare/next-on-pages': minor
+---
+
+Fix buildSuccess check that keeps users from receiving useful information.
```
Comment (dario-piotrowicz) [inline .changeset/modern-keys-mix.md:5]:
```suggestion
---
'@cloudflare/next-on-pages': patch
---

prevent buildSuccess check from crushing with a bad error message
```
this is fixing a bug so it should be a patch I think 🙂 

moreover the message can be improved (sorry I think you followed the initial title of the issue that I wrote but I got it a bit off, that's my bad)

**Review by dario-piotrowicz — COMMENTED** [review]
Thanks a bunch @abhishek276533 😄 

just one small correction on the changeset 🙏

**Comment by dario-piotrowicz** [comment]
@abhishek276533 also the linting is failing, could you please run `npm run prettier:fix` to fix it? 🙂

File: .changeset/modern-keys-mix.md (hunk 1 above)
Comment (abhishek276533) [reply .changeset/modern-keys-mix.md:5]: yeah, I just followed the initial title. No worries have added the change
image

**Review by dario-piotrowicz — APPROVED** [review]
Thanks @abhishek276533 😃

File: .changeset/silly-tips-sparkle.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@cloudflare/next-on-pages': patch
+---
+
+prevent buildSuccess check from crushing with a bad error message.
```
Comment (james-elicx) [inline .changeset/silly-tips-sparkle.md:5]:
```suggestion
prevent buildSuccess check from crashing with a bad error message.
```

**Pull request merged**